test_scripts/API/ButtonSubscription/common_buttonSubscription.lua

```diff
@@ -0,0 +1,378 @@
+---------------------------------------------------------------------------------------------------
+-- Common module
+---------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local test = require("user_modules/dummy_connecttest")
+local events = require('events')
+
+
+--[[ Variables ]]
+local m = actions
+m.hashId = {}
+m.wait = utils.wait
+m.cloneTable = utils.cloneTable
+
+m.buttons = {
+  "OK",
+  "PLAY_PAUSE",
+  "SEEKLEFT",
+  "SEEKRIGHT",
+  "TUNEUP",
+  "TUNEDOWN",
+  "PRESET_0",
+  "PRESET_1",
+  "PRESET_2",
+  "PRESET_3",
+  "PRESET_4",
+  "PRESET_5",
+  "PRESET_6",
+  "PRESET_7",
+  "PRESET_8",
+  "PRESET_9",
+  "SEARCH"
+}
+m.media_buttons = {
+  "PLAY_PAUSE",
+  "SEEKLEFT",
+  "SEEKRIGHT",
+  "TUNEUP",
+  "TUNEDOWN",
+  "PRESET_0",
+  "PRESET_1",
+  "PRESET_2",
+  "PRESET_3",
+  "PRESET_4",
+  "PRESET_5",
+  "PRESET_6",
+  "PRESET_7",
+  "PRESET_8",
+  "PRESET_9"
+}
+
+m.errorCode = {
+  "UNSUPPORTED_REQUEST",
+  "DISALLOWED",
+  "REJECTED",
+  "IN_USE",
+  "DATA_NOT_AVAILABLE",
+  "TIMED_OUT",
+  "INVALID_DATA",
+  "CHAR_LIMIT_EXCEEDED",
+  "INVALID_ID",
+  "DUPLICATE_NAME",
+  "APPLICATION_NOT_REGISTERED",
+  "OUT_OF_MEMORY",
+  "TOO_MANY_PENDING_REQUESTS",
+  "GENERIC_ERROR",
+  "USER_DISALLOWED",
+  "TRUNCATED_DATA",
+  "READ_ONLY"
+}
+
+--[[ Common Functions ]]
+--[[ @rpcSuccess: performs button Subscrioption and Unsubscription with SUCCESS resultCode
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! pRpc: RPC name
```

Comment:
use the same separator as above:
--! pAppId - application number (1, 2, etc.)
--! pRpc **-** RPC name

Perform update through all file